tests/threadpooltest.py

```diff
@@ -131,15 +131,15 @@ def testConcurrency(self):
         def accounting(i):
             counts[thread.get_ident()] += 1
         def killit(i):
-            for j in xrange(N):
+            for j in range(N):
                 self.pool.apply_async(accounting, (i,))
-        threads = [ Thread(target=killit, args=(i,)) for i in xrange(M) ]
+        threads = [ Thread(target=killit, args=(i,)) for i in range(M) ]
         for t in threads:
             t.start()
         for t in threads:
             t.join()
         self.join_continue(self.pool, 60)
-        total_counts = self.pool.apply(sum, (counts.itervalues(),))
+        total_counts = self.pool.apply(sum, (iter(counts.values()),))
```

Comment:
Don't `values` already return an iterator?

Reply:
Only if they're dicts. This was done by 2to3 and I bet it didn't know defaultdict acts like a dict in this case.